selfdrive/car/toyota/carcontroller.py

```diff
@@ -142,8 +142,8 @@ def update(self, CC, CS, now_nanos):
 
       # Press distance button until we are at the correct bar length. Only change while enabled to avoid skipping startup popup
       if self.frame % 6 == 0:
-        if CS.pcm_follow_distance_values.get(CS.pcm_follow_distance, "UNKNOWN") != "FAR" and CS.out.cruiseState.enabled and \
-          self.CP.carFingerprint not in UNSUPPORTED_DSU_CAR:
+        desired_distance = 4 - hud_control.leadDistanceBars
+        if CS.pcm_follow_distance != desired_distance and CS.out.cruiseState.enabled and self.CP.carFingerprint not in UNSUPPORTED_DSU_CAR:
```

Comment:
Honestly I think the vehicle should determine this, and openpilot should just follow the vehicle's behaviour (e.g. vehicle boots up in 3 bar, openpilot starts up in relaxed mode). Toyota vehicles default to 3 bars, and the user can just press the button once or twice to get into the desired follow distance. It will also be a lot easier to implement from a code perspective.

Reply:
As I noted above, Toyota is the odd one out in that the PCM manages the following distance. Ford, Hyundai, Honda, GM (and I assume Subaru, Chrysler, Tesla) all have clusters that can be sent a number of bars/distance.

If we had the car manage it, on cars with four bars the experience would be pretty weird as two would have no effect. Plus, we can't do this on cars we knock out the ACC module like Hyundai CAN FD, and need to conditionally manage state in CarState.

Having the state managed by the personality param + buttons seems to be the easiest to implement across all brands.